server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -733,11 +762,24 @@ public void run()
       catch (Exception e) {
         log.makeAlert(e, "Caught exception, ignoring so that schedule keeps going.").emit();
       }
-      finally {
-        if (balancerExec != null) {
-          balancerExec.shutdownNow();
```

Comment:
do we need to worry about abandoning resources after removing this? Should the exec possibly be shutdown in stopBeingLeader() in your patch?